### PR TITLE
publiccloud: Add ability to skip cleanup of created env

### PR DESCRIPTION
**Pull request by asmorodskyi**
Add ability to skip cleanup of created env

### CODE REVIEW

**Comment by asmorodskyi** [comment]
@pdostal , @cfconrad - WDYT ?

File: lib/publiccloud/basetest.pm [hunk 1]
```diff
@@ -80,10 +80,13 @@ sub _cleanup {
...
-        !($self->{result} eq 'fail' && $flags->{fatal})) {
-        return;    # skip cleanup
-    }
+    # currently we have two cases when cleanup of image will be skipped:
+    # 1. Calling module needs to have publiccloud_multi_module => 1 test flag
+    # and not have fatal => 1. Job should not have result = 'fail'
+    # 2. Job should have PUBLIC_CLOUD_NO_CLEANUP defined and job should have result = 'fail'
+    return if (($flags->{publiccloud_multi_module} && !($self->{result} eq 'fail' && $flags->{fatal})) ||
+        ($self->{result} eq 'fail' && get_var('PUBLIC_CLOUD_NO_CLEANUP')));
+
```
Comment (cfconrad) [inline lib/publiccloud/basetest.pm:89]:
```suggestion
    # currently we have two cases when cleanup of image will be skipped:
    # 1. Calling module needs to have publiccloud_multi_module => 1 test flag
    # and not have fatal => 1. Job should not have result = 'fail'
    if ($flags->{publiccloud_multi_module} &&
        !($self->{result} eq 'fail' && $flags->{fatal})) {
        return;    # skip cleanup
    }
    # 2. Job should have PUBLIC_CLOUD_NO_CLEANUP defined and job should have result = 'fail'
    return if ($self->{result} eq 'fail' && get_var('PUBLIC_CLOUD_NO_CLEANUP')));
```

**Comment by asmorodskyi** [comment]
VR : http://autobot.suse.de/tests/642 - w/o PUBLIC_CLOUD_NO_CLEANUP

**Comment by cfconrad** [comment]
LGTM, lets wait for travis and VR's

**Comment by asmorodskyi** [comment]
VR : http://autobot.suse.de/tests/643 - PUBLIC_CLOUD_NO_CLEANUP_ON_FAILURE = 1

**Pull request merged**